crates/core/src/config.rs

```diff
@@ -42,62 +38,334 @@ const QUALIFIER: &str = "";
 const ORGANIZATION: &str = "The Freenet Project Inc";
 const APPLICATION: &str = "Freenet";
 
+#[derive(clap::Parser, Debug, Serialize, Deserialize)]
+pub struct ConfigArgs {
+    /// Node operation mode.
+    #[clap(value_enum, env = "MODE")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mode: Option<OperationMode>,
+
+    /// Path to the configuration file.
+    #[clap(short, long)]
+    #[serde(skip)]
+    pub config: Option<PathBuf>,
+
+    #[clap(flatten)]
+    #[serde(flatten)]
+    pub gateway: GatewayArgs,
```

Comment:
what is this being used for? is the http gateway server? if so let's call it like that, otherwise can be confused with "gateway node"

Reply:
Yes, it is for http gateway server